napari/utils/migrations.py

```diff
@@ -137,6 +146,13 @@ def _setter(instance, value: Any) -> None:
         warnings.warn(msg, category=FutureWarning, stacklevel=3)
         setattr(instance, new_name, value)
 
+    if not hasattr(obj, '_deprecated_properties_list'):
+        obj._deprecated_properties_list = []
+
+    obj._deprecated_properties_list.append(
```

Comment:
@Czaki : one problem with this approach is that it cannot handle deprecated properties (e.g. `current_edge_color` / `current_border_color`) that are not part of the state (i.e. returned by `get_state`).

I think that should be pretty easy with some modifications. Let me try it.

Reply:
Why is it not part of state? It is deprecated using `add_deprecated_property` that updates this attribute.

Reply:
I'm not certain why some of them are not state, but I think we would want the flexibility to have deprecated properties that are not state.